### PR TITLE
Fix for atomicv on CXI

### PR DESCRIPTION
**Pull request by wrrobin**

### CODE REVIEW

File: src/shmem_comm.h [hunk 1]
```diff
@@ -336,6 +336,7 @@ shmem_internal_atomicv(shmem_ctx_t ctx, void *target, const void *source,
         shmem_internal_fetch_atomic(ctx, ((uint8_t *) target) + (i * type_size),
                                     ((uint8_t *) source) + (i * type_size), &tmp_fetch, type_size,
                                     pe, op, datatype);
+        shmem_transport_get_wait((shmem_transport_ctx_t *)ctx);
```
Comment (davidozog) [inline src/shmem_comm.h:339]: Can we move this outside the loop over `count`?  I don't quite see why we need to wait after every atomic?

File: src/shmem_comm.h [hunk 2]
```diff
@@ -336,6 +336,7 @@ shmem_internal_atomicv(shmem_ctx_t ctx, void *target, const void *source,
...
 #ifdef SOS_OFI_CXI_QUIET_RATE
```
Comment (davidozog) [inline src/shmem_comm.h:340]: Is there any chance we can remove this after the above fix?

File: src/shmem_comm.h (hunk 1 above)
Comment (wrrobin) [reply src/shmem_comm.h:339]: Nope, we cant as the loop itself can go beyond the number of atomics that causes the issue on CXI.

File: src/shmem_comm.h (hunk 2 above)
Comment (wrrobin) [reply src/shmem_comm.h:340]: I think based on the atomic OP, we cant get rid of this completely. But, will check.

**Review by davidozog — APPROVED** [review]
As discussed offline, perhaps `SHMEM_REDUCE_ALGORITHM=recdbl` (maybe `ring`?) is a better workaround (when disabling non-fetching AMO) given this could dramatically impair the performance of `linear` and `tree` algorithms.

**Pull request merged**